### ISSUE TITLE
Ability to disable global keyboard shortcuts for a specific app

### ISSUE THREAD

**Issue by fatlazycat** [pqrs-org/Karabiner-Elements#2457]
Is there anyway to disable all global keyboard shortcuts so a named specific app can get them the active ?

Problem is using an app like IntelliJ IDEA and many conflicts.

Thanks

**Comment by Bennett-Lynch** [comment]
+1. I love Karabiner and I use most of the PC-Style shortcuts, but for IntelliJ IDEA, I want to use the built-in XWin keymap and not rely on Karabiner (other than perhaps [swapping control/command](https://stackoverflow.com/questions/35874608/intellij-default-windows-keymap-on-mac-os-x)). Currently, it seems like the only solution is to exclude Intellij from the N number of rules.

**Comment by fatlazycat** [comment]
Ah, to be clearer I wanted to stop global keyboard shortcuts defined in macOS or other apps from being activated for IntelliJ.

When I invoke some shortcut in IntelliJ it then also invokes the global shortcut as well and I have to hunt down where it is defined in the other app and turn it off.

**Comment by puresick** [comment]
I would also be interested in some similar feature. In my case, I would love to have the shortscuts in Karabiner be disabled if I focused Alacritty or any other terminal application.

**Comment by cppguru** [comment]
+1.  The default MacOS shortcuts are a pain when working on Windows using RDP or a Virtual Machine.  VMs may give the ability to hijack keys, but then one has to configure that for each, individually.  Otherwise I could just use Karabiner to have a set of "Windows unfriendly hotkeys to turn off" and set the apps to which it applies.

**Comment by cesalberca** [comment]
Is there a workaround for this?

**Comment by JoeBrig** [comment]
I use VIM style remaps of arrow keys in MacOS. This causes problems when working in the VIM app via a terminal. The ability to turn off mappings for specific apps, Alacritty in my case, would help.

**Comment by exsesx** [comment]
Yeah, it's a nice feature to have. I use IDE for coding, and I want to use Control+Space for IDE key bindings and not switch languages in macOS...

**Comment by top-kat** [comment]
Any news on this ?

**Comment by naimmas** [comment]
Any updates? It would be a great feature especially when working on Win RDP.

**Comment by KaKi87** [comment]
+1

**Comment by mlhktp** [comment]
+1

**Comment by tekezo** [comment]
In the latest Karabiner-Elements beta, I've added to.from_event, which makes it possible to implement a Pass Through Mode that disables all Complex Modifications under specific conditions.
https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to/from-event/

How to upgrade Karabiner-Elements to the latest beta version:
<https://karabiner-elements.pqrs.org/docs/manual/operation/check-for-updates/#using-the-latest-beta-version>

**Comment by fatlazycat** [comment]
Thanks
